### PR TITLE
Itertools takewhile

### PR DESCRIPTION
**Pull request by Jongy**
After https://github.com/RustPython/RustPython/pull/949

### CODE REVIEW

**Comment by Jongy** [comment]
Seems like this needs to be rebased...

File: vm/src/stdlib/itertools.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        if *self.stop_flag.borrow() {
+            return Err(new_stop_iteration(vm));
+        }
+
+        let obj = call_next(vm, &self.iterable);
```
Comment (windelbouwman) [inline vm/src/stdlib/itertools.rs]: Please use the `?` operator here, like this:
```rust
let obj = call_next(vm, &self.iterable)?;
```
The match statement can be replaced by the `?` suffix.

File: vm/src/stdlib/itertools.rs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            Ok(verdict) => match objbool::boolval(vm, verdict) {
+                Ok(true) => Ok(obj),
+                Ok(false) => {
+                    *self.stop_flag.borrow_mut() = true;
+                    Err(new_stop_iteration(vm))
+                }
+                Err(err) => Err(err),
+            },
+            Err(err) => Err(err),
+        }
```
Comment (windelbouwman) [inline vm/src/stdlib/itertools.rs]: Please replace the above by this:
```rust
let verdict = vm.invoke(self.predicate.clone(), vec![obj.clone()])?;
let verdict = objbool::boolval(vm, verdict)?;
if verdict {
  Ok(obj)
} else {
                    *self.stop_flag.borrow_mut() = true;
                    Err(new_stop_iteration(vm))
}

**Comment by Jongy** [comment]
> Please check my comments about the usage of the ? operator for error handlings. It can be applied at multiple places in your change and makes the code much more compact.

Hmm, cool. Didn't know of this one. I'm learning Rust on the way :)

Thanks for the instructive comments.

**Pull request merged**